plugins/obs-filters/nvafx-load.h

```diff
@@ -0,0 +1,159 @@
+typedef void *NvAFX_Handle;
+
+typedef NvAFX_Status
+	NVAFX_API (*NvAFX_GetEffectList_t)(int *num_effects,
+					   NvAFX_EffectSelector *effects[]);
+typedef NvAFX_Status
+	NVAFX_API (*NvAFX_CreateEffect_t)(NvAFX_EffectSelector code,
+					  NvAFX_Handle *effect);
+typedef NvAFX_Status NVAFX_API (*NvAFX_DestroyEffect_t)(NvAFX_Handle effect);
+typedef NvAFX_Status
+	NVAFX_API (*NvAFX_SetU32_t)(NvAFX_Handle effect,
+				    NvAFX_ParameterSelector param_name,
+				    unsigned int val);
+typedef NvAFX_Status
+	NVAFX_API (*NvAFX_SetString_t)(NvAFX_Handle effect,
+				       NvAFX_ParameterSelector param_name,
+				       const char *val);
+typedef NvAFX_Status NVAFX_API (*NvAFX_SetFloat_t)(
+	NvAFX_Handle effect, NvAFX_ParameterSelector param_name, float val);
+typedef NvAFX_Status
+	NVAFX_API (*NvAFX_GetU32_t)(NvAFX_Handle effect,
+				    NvAFX_ParameterSelector param_name,
+				    unsigned int *val);
+typedef NvAFX_Status
+	NVAFX_API (*NvAFX_GetString_t)(NvAFX_Handle effect,
+				       NvAFX_ParameterSelector param_name,
+				       char *val, int max_length);
+typedef NvAFX_Status NVAFX_API (*NvAFX_GetFloat_t)(
+	NvAFX_Handle effect, NvAFX_ParameterSelector param_name, float *val);
+typedef NvAFX_Status NVAFX_API (*NvAFX_Load_t)(NvAFX_Handle effect);
+typedef NvAFX_Status NVAFX_API (*NvAFX_Run_t)(NvAFX_Handle effect,
+					      const float **input,
+					      float **output,
+					      unsigned num_samples,
+					      unsigned num_channels);
+
+static NvAFX_GetEffectList_t NvAFX_GetEffectList = NULL;
+static NvAFX_CreateEffect_t NvAFX_CreateEffect = NULL;
+static NvAFX_DestroyEffect_t NvAFX_DestroyEffect = NULL;
+static NvAFX_SetU32_t NvAFX_SetU32 = NULL;
+static NvAFX_SetString_t NvAFX_SetString = NULL;
+static NvAFX_SetFloat_t NvAFX_SetFloat = NULL;
+static NvAFX_GetU32_t NvAFX_GetU32 = NULL;
+static NvAFX_GetString_t NvAFX_GetString = NULL;
+static NvAFX_GetFloat_t NvAFX_GetFloat = NULL;
+static NvAFX_Load_t NvAFX_Load = NULL;
+static NvAFX_Run_t NvAFX_Run = NULL;
+
+void release_lib(void)
+{
+	NvAFX_GetEffectList = NULL;
+	NvAFX_CreateEffect = NULL;
+	NvAFX_DestroyEffect = NULL;
+	NvAFX_SetU32 = NULL;
+	NvAFX_SetString = NULL;
+	NvAFX_SetFloat = NULL;
+	NvAFX_GetU32 = NULL;
+	NvAFX_GetString = NULL;
+	NvAFX_GetFloat = NULL;
+	NvAFX_Load = NULL;
+	NvAFX_Run = NULL;
+	if (nv_audiofx) {
+		FreeLibrary(nv_audiofx);
+		nv_audiofx = NULL;
+	}
+}
+
+static void nvafx_get_sdk_path(char buffer[MAX_PATH])
+{
+	char value[MAX_PATH];
+	PVOID pvData = value;
+	DWORD BufferSize = 8192;
+	LPDWORD pcbData;
+	LSTATUS status = RegGetValue(
+		HKEY_LOCAL_MACHINE,
+		TEXT("SYSTEM\\CurrentControlSet\\Control\\Session Manager\\Environment"),
+		TEXT("NVAFX_SDK_DIR"), RRF_RT_REG_SZ, NULL, pvData,
+		&BufferSize);
+	wcstombs(buffer, (LPCWSTR)pvData, MAX_PATH);
+	if (status != ERROR_SUCCESS) {
+		printf("[noise suppress: NVAFX redist is not installed]");
+		return NULL;
+	} else {
+		printf("[noise suppress: NVAFX SDK redist path is %s]", buffer);
+	}
```

Comment:
I don't know if this is your code, but you should not call `wcstombs()` before checking for an error. It should be within the else value. Additionally, you should not use `wcstombs()`. You should use `os_wcs_to_utf8()`.

Reply:
fixed.
Additionally I've moved the log lines to the main filter code to make use of do_log functions.